screens/AddPlaceScreen.js

```diff
@@ -0,0 +1,233 @@
+import React, { useState } from 'react'
+import { View, StyleSheet, TextInput, Alert } from 'react-native'
+import Place from '../models/place'
+import { Ionicons } from '@expo/vector-icons'
+import styled from "styled-components/native"
+import { LOCATION_API } from 'react-native-dotenv'
+
+import PLACES from '../data/dummy-data'
+
+const fetchLocationData = async (street, city) => {
+  let search = `${street} ${city} `
+  const res = await fetch(`https://eu1.locationiq.com/v1/search.php?key=${LOCATION_API}&q=${search}&format=json`)
+  const json = await res.json()
+  return json
+}
+
+export const AddPlaceScreen = (navData) => {
+
+  const [title, setTitle] = useState("")
+  const [description, setDescription] = useState("")
+  const [street, setStreet] = useState("")
+  const [city, setCity] = useState("Stockholm")
+
+  let lastId = PLACES.slice(-1)[0].id
+
+
+  const saveData = () => {
+    fetchLocationData(street, city)
+      .then((result) => {
+
+        if (title == "" || street == "" || city == "") {
+          Alert.alert("Fill in title, street and city")
+          return
+        }
+        PLACES.push(new Place(lastId + 1, title, description, { 'latitude': Number(result[0].lat), 'longitude': Number(result[0].lon) }, { 'street': street, 'city': city }))
+        Alert.alert('Place saved!')
+        navData.navigation.navigate("Map", { pl: PLACES })
+
+      })
+  }
+
+  const clickedIcon = () => {
+    alert("Button pressed")
+  }
+
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.form}>
+        <TextInput
+          autoCorrect={false}
+          style={styles.formInput}
+          placeholder={'Title'}
+          onChangeText={text => {
+            setTitle(text)
+          }}
+        />
+        <TextInput
+          autoCorrect={false}
+          style={styles.formInput}
+          placeholder={'Description'}
+          onChangeText={text => {
+            setDescription(text)
+          }}
+        />
+
+        <TextInput
+          autoCorrect={false}
+          style={styles.formInput}
+          placeholder={'Street'}
+          onChangeText={text => {
+            setStreet(text)
+          }}
+        />
+
+        <TextInput
+          autoCorrect={false}
+          style={styles.formInput}
+          value={city}
+          placeholder={'City'}
+          onChangeText={text => {
+            setCity(text)
+          }}
+        />
+
+
+        <View style={styles.iconsContainer}>
+          <IconButton>
+            <Ionicons
+              color={"#fff"}
+              style={styles.icon}
+              name="ios-home"
+            />
+          </IconButton>
```

Comment:
You could make this into its own component and have props like name and color. Where you also have a default color to white so you don't have to specify everytime.
`<Icon name="ios-home" /> `and then just add this for each icon button